src/plugins/data_table/data_table.js

```diff
@@ -261,24 +291,177 @@ var LocalDataTable = (function() {
       this.trigger("change:selected", data);
     },
 
+    // DataTables does not provide a good way to programmatically disable sorting, so we:
+    // 1) remove the default sorting event handler that dataTables adds
+    // 2) Create a div and put the header in it.  We need to do this so sorting doesn't conflict with filtering
+    // on the click events.
+    // 3) insert our own event handler on the div that stops the event if we are locked
+    // 4) re-insert the dataTables sort event handler
     _installSortInterceptors: function() {
-      // dataTables does not provide a good way to programmatically disable sorting, so we:
-      // 1) remove the default sorting event handler that dataTables adds
-      // 2) insert our own that stops the event if we are locked
-      // 3) re-insert the dataTables sort event handler
       var self = this;
       this.dataTable.find("thead th").each(function(index) {
-        $(this).off("click.DT").on("click", function(event) {
+        $(this).off("click.DT");
+        // put the header text in a div
+        var nDiv = document.createElement('div');
+        nDiv.className = "DataTables_sort_wrapper";
+        $(this).contents().appendTo(nDiv);
+        this.appendChild(nDiv);
+        // handle clicking on div as sorting
+        $('.DataTables_sort_wrapper', this).on("click", function(event) {
           if (self.lock("sort")) {
             event.stopImmediatePropagation();
           }
         });
         // default sort handler for column with index
-        self.dataTable.fnSortListener($(this), index);
+        self.dataTable.fnSortListener($('.DataTables_sort_wrapper', this), index);
+      });
+    },
+
+
+    // Here we make different controls based on the filter type we're dealing with.
+    // * string filtering requires a single text input
+    // * numeric filtering requires three text inputs for greater than, less than, and
+    //   equal to.  these text inputs need to be uniquely identified so we can filter on
+    //   values entered into more than one of them
+    // * list filtering requires a list of labeled checkboxes for each filter option
+    //   these checkboxes need to be identified by the value they represent
+    // The IDs "value", "gt", "lt" and "eq" are used to determine in what element in the
+    // filter object in the column manager we store the value entered by the user
+    _generateFilteringControls: function(head, col) {
+      var table = this;
+      var filter = col.filter;
+      if (filter.type == "string") {
+        $(head).append(table.filterStringTemplate( { title: col.title } ));
+      } else if (filter.type == "numeric") {
+        $(head).append(table.filterNumericTemplate());
+      } else if (filter.type == "list") {
+        var checkList = '<ul>';
+        for (var i = 0; i < filter.options.length; i++)
```

Comment:
That's a good idea.  Ideally it should never happen because it should throw an error on the backend, but I don't know that for certain and it's always good to be safe.  Will take care of that.

Do you want to throw an exception or do you want to just not make the filtering controls if filter.options is undefined when filter.type === 'list'?